DBSCE/KeyGeneration.md

```diff
@@ -0,0 +1,91 @@
+@="progid:App.Component.1.0"
+
+[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Edge\LocalKeyHelperList\login.fabrikam.com]
+@="clsid:3AD05D77-6885-4BAF-A727-D30690A2A28A"
+```
+
+Currently the supported schemes are:
+
+1. clsid: \<GUID\> - CLSID of a COM class that implements the Local Key Helper.
+2. progid:\<ProgId string\> - [a programic identifier](https://learn.microsoft.com/en-us/windows/win32/com/-progid--key) of the local key helper.
+
+This scheme can be extended by other schemes in future.
+The local key helper registry key can have extra values, which we can use for in future.
+
+Here is a visual example how Local key helper is organized:
+
+![Local key helper registry key](./images/keyhelper-reg.png)
+
+When the browser needs to communicate with a local key helper. It uses its ID to locate the registry key, then reads the default value of this registry key and activates the object by the activation scheme (CLSID or ProgId according to the document). After activation it queries ILocalKeyHelper interface and invokes corresponding methods.
+
+#### Well-known local key helpers on Windows
+
+A well-known local key helper list on Windows is a list of helperId to clsid, that either harcoded or predefined in the browser settings.
+
+### **Attestation Service**:
+
+A service that is responsible for verifying the device registration and providing the attestation to the IdP. The attestation service can be owned by the IdP or a third party. Before defining the attestation service, let us get familiar with the building blocks of binding and proof of binding.
+
+There are three artifacts that are generated during/after the device registration process, which are used to prove the device binding. These are the _binding key_, the _attestation key_, and the _binding statement_.
+
+#### **Binding Key**:
+
+A _binding key_ is an asymmetric key pair that is used to bind an auth cookie.It is identified by a key ID and it is the responsibility of the browser to remember which key ID corresponds to which Local Key Helper and to use it for DBSC signatures and key management. As there could be multiple `devices` on a single physical device and or many device registration clients on a single device (mentioned [above](#device-registration-client)), the key mapping is expected to be managed by the Browser and the Local Key Helper.
+
+This **binding key** is the same as defined in the DBSC proposal [here](https://github.com/WICG/dbsc?tab=readme-ov-file#maintaining-a-session) and is expected to be created in a secure enclave (TPM or Keyguard) on the original device. However, please note that in the context of the DBSC proposal, the binding key validation is not guaranteed to be attack free, as it can be generated by malware running on the device. In the context DBSC(E), the binding key is generated during the device registration process, and hence is guaranteed to be attack free if generated in a clean room environment.
```

Comment:
>> In the context DBSC(E), the binding key is generated during the device registration process, and hence is guaranteed to be attack free if generated in a clean room environment.

It's not entirely correct that binding key is generated during device registration. We generate attestation key during device registration, such as we can cryptographically attest binding key at a later point when it's generated. I'd suggest rewording to avoid confusion.

Reply:
Reworded. This section is also moved to the main document, please check there.